pkg/reporter/base.go

```diff
@@ -161,6 +177,65 @@ func (o RegulaOutput) AggregateByFilepath() ResultsByFilepath {
 	return byFilepath
 }
 
+// AggregateByRule returns all rule results grouped by rule
+func (o RegulaOutput) AggregateByRule() ResultsByRule {
+
+	// Collect all results by rule ID
+	byRule := map[string][]*RuleResult{}
```

Comment:
Sorry I didn't think of this earlier, but maybe we should do this by rule name rather than ID. Rule IDs are optional, so this might produce strange output for custom rules without IDs or metadocs. Rule names are inferred from the package name, so they should be unique and present for all rules.

Reply:
Done